ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -1237,6 +1240,32 @@ public void getIndexSegmentFilesForLogSegmentTest() {
     }
   }
 
+  /**
+   * Tests {@link PersistentIndex#close()} can correctly cancel the scheduled persistor task and makes sure no persistor
+   * is running background after index closed.
+   * @throws IOException
+   * @throws StoreException
+   */
+  @Test
+  public void closeIndexToCancelPersistorTest() throws StoreException, InterruptedException {
+    state.index.close();
+    // re-initialize index by using mock scheduler (the intention is to speed up testing by using shorter period)
+    MockScheduledExecutorService mockScheduler =
+        new MockScheduledExecutorService(1, false, 0, 100, TimeUnit.MILLISECONDS);
+    mockScheduler.setContinueExistingPeriodicTasksAfterShutdownPolicy(false);
+    mockScheduler.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
```

Comment:
I wonder if we could use a mockito spy to eliminate the duplicated code: `mockito.spy(Utils.newScheduler(...))`

https://www.baeldung.com/mockito-spy

Reply:
done